src/boot/efifirmware.h

```diff
@@ -0,0 +1,27 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+#pragma once
+
+#include "efi.h"
+
+#define FWHEADERMAGIC 0xfeeddead
+
+/* The structure of the efifw UKI blob is the following:
+ *  ----------------------------------
+ *  EfiFw header|metadata|fwid|payload
+ *  ----------------------------------
+ *  The header defines the length of metadata, fwid and payload.
+ *  The fwid is a NULL terminated string.
+ *  The payload contains the actual efi firmware.
+ */
+typedef struct EfiFwHeader {
+        uint32_t magic;
+        uint32_t total_size; /* includes header, fwid and payload */
+        uint32_t fw_metadata_offset;
+        uint32_t fw_metadata_len;
+        uint32_t fwid_offset;
+        uint32_t fwid_len;
+        uint32_t payload_offset;
+        uint32_t payload_len;
+} EfiFwHeader;
```

Comment:
who defined that structure? is that by us? or by efi firmware?

Reply:
why does this need offset and len for each of the parts? seems len should suffice?

Reply:
We are defining it. Its up to us. No standards.

Reply:
then simplify this, i.e. just include a magic, a header length, an fwid string length and a payload length. if we later want to add more fields we'd increase the header length. and the total length is always the length of header + fwid + payload.

Reply:
I wanted to have  field for the firmware metadata just in case.

Reply:
we can extend the header when the time comes. if we have a header size then we can add additional fields any time we like.